Index: gendarme.xsd
===================================================================
--- gendarme.xsd	(revisión: 104097)
+++ gendarme.xsd	(copia de trabajo)
@@ -28,6 +28,7 @@
 		<xs:attribute name="include" type="notEmptyString" use="required"/>
 		<xs:attribute name="exclude" type="notEmptyString"/>
 		<xs:attribute name="from" type="notEmptyString" use="required"/>
+		<xs:attribute name="applyTo" type="visibilityModifier"/>
 	</xs:complexType>
 
 	<xs:complexType name="parameterType">
@@ -44,4 +45,12 @@
 		</xs:restriction>
 	</xs:simpleType>
 
+	<xs:simpleType name="visibilityModifier">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="all"/>
+			<xs:enumeration value="visible"/>
+			<xs:enumeration value="nonvisible"/>
+		</xs:restriction>
+	</xs:simpleType>
+
 </xs:schema>
Index: ChangeLog
===================================================================
--- ChangeLog	(revisión: 104097)
+++ ChangeLog	(copia de trabajo)
@@ -1,3 +1,8 @@
+2008-05-27  Nestor Salceda  <nestor.salceda@gmail.com>
+
+	* gendarme.xsd: Added the applyTo section for allowing to apply rules to
+	different visibility scenarios.
+
 2008-05-25  Sebastien Pouliot  <sebastien@ximian.com>
 
 	* XmlResultWriter.cs: Fix sorting
