Index: ChangeLog
===================================================================
--- ChangeLog	(revisión: 107968)
+++ ChangeLog	(copia de trabajo)
@@ -1,3 +1,8 @@
+2008-07-15  Nestor Salceda  <nestor.salceda@gmail.com>
+
+	* ModuleRocks.cs: Look up the Mono.Cecil.Mdb.dll in the GAC if can't be
+	found in the current folder.
+
 2008-07-12  Sebastien Pouliot  <sebastien@ximian.com>
 
 	* TypeRocks.cs: Added IsStatic for static types (2.0+). Fixed 
Index: ModuleRocks.cs
===================================================================
--- ModuleRocks.cs	(revisión: 107968)
+++ ModuleRocks.cs	(copia de trabajo)
@@ -85,7 +85,9 @@
 			// so we start by looking for it's debugging symbol file
 			if (File.Exists (symbol_name)) {
 				// "always" if we can find Mono.Cecil.Mdb
-				reader_type = Type.GetType ("Mono.Cecil.Mdb.MdbFactory, Mono.Cecil.Mdb");
+				reader_type = Type.GetType ("Mono.Cecil.Mdb.MdbFactory, Mono.Cecil.Mdb, Version=0.2.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756");
+				// load the assembly from the current folder if
+				// it is here, or fallback to the gac
 			}
 			
 			// if we could not load Mono's symbols then we try, if not running on Mono,
